preliminary work for projection changes

Berthalamew opened the pull request:
changes and additions to prepare for projection work

Berthalamew committed:
fix bug in updating the rasterizer constant cache

would loop for the incorrect amount of times when copying over contents that don't match

Berthalamew committed:
update motd code to put the png in the relative path

Berthalamew committed:
update render_lod_new code

- name the change color function and properly label the render_lighting member

Berthalamew committed:
rename k_real_math_epsilon to k_real_epsilon

Berthalamew committed:
fix bug in matrix4x3_inverse

fix typo when calculating the y position, function would multiply output->vectors.left.j by inverse_pos.x rather than inverse_pos.y previously.

Berthalamew committed:
make cubemap_target_size_get a public function

Berthalamew committed:
add matrix4x3_transform_normal and matrix4x3_transform_plane

Berthalamew committed:
math changes

- add VALIDATE_REAL_ENABLED preprocessor constant
- add macros: assert_valid_real and valid_real_matrix4x3
- add functions: valid_real_point3d, valid_real_vector3d_axes3, valid_real_matrix4x3, negate_vector3d, and scale_vector4d
- adjust parameter names in cross_product3d

Berthalamew committed:
remove unused includes

nukeulater commented on xlive/Blam/Engine/math/matrix_math.cpp:
```diff
@@ -130,48 +130,45 @@ void matrix4x3_from_point_and_vectors(real_matrix4x3* matrix, const real_point3d
 }
 
-void matrix4x3_inverse(const real_matrix4x3* input, real_matrix4x3* output)
+real_matrix4x3* matrix4x3_inverse(const real_matrix4x3* input, real_matrix4x3* output)
```
checked this function, in this instance the out vector pointer is not actually returned

Berthalamew replied:
removed the return value

nukeulater commented on xlive/Blam/Engine/rasterizer/dx9/rasterizer_dx9.h:
```diff
@@ -50,10 +50,9 @@ class c_rasterizer_constant_4f_cache
                 if (count != i)
                 {
-                    for (int32 j = i + index; j < size; j++)
+                    for (i; i < count; ++i)
```
you can remove the first expression in this
i.e `for (; i < count; ++i)`

and `if (count != i)` could also be removed

Berthalamew replied:
Changed

Berthalamew committed:
add requested changes

- remove return value from matrix4x3_inverse
- remove unneeded if statement from c_rasterizer_constant_4f_cache::test_cache
